code/modules/research/server.dm

```diff
@@ -187,3 +196,112 @@
 	playsound(src, "sparks", 75, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
 	obj_flags |= EMAGGED
 	to_chat(user, span_notice("You disable the security protocols."))
+
+/// Master R&D server. As long as this still exists and still holds the HDD for the theft objective, research points generate at normal speed. Destroy it or an antag steals the HDD? Half research speed.
+/obj/machinery/rnd/server/master
+	var/obj/item/computer_hardware/hard_drive/cluster/hdd_theft/source_code_hdd
+	var/deconstruction_state = HDD_PANEL_CLOSED
+	var/front_panel_screws = 4
+	var/hdd_wires = 6
+
+/obj/machinery/rnd/server/master/Initialize()
+	. = ..()
+	name = "\improper Master " + name
+	source_code_hdd = new(src)
+	SSresearch.master_servers += src
+
+	add_overlay("RD-server-objective-stripes")
+
+/obj/machinery/rnd/server/master/Destroy()
+	if(source_code_hdd)
+		QDEL_NULL(source_code_hdd)
+
+	SSresearch.master_servers -= src
+
+	return ..()
+
+/obj/machinery/rnd/server/master/attackby(obj/item/attacking_item, mob/user, params)
+	if(!source_code_hdd)
+		return ..()
+
+	switch(deconstruction_state)
+		if(HDD_PANEL_CLOSED)
+			if(istype(attacking_item, /obj/item/screwdriver/hdd_extraction))
+				to_chat(user, span_notice("Your [attacking_item.name] fits perfectly into the slots of [src]'s front panel screws. You can see [front_panel_screws] screws. You start unscrewing them..."))
+				while(attacking_item.use_tool(src, user, 100, volume=100))
+					front_panel_screws--
+
+					if(front_panel_screws <= 0)
+						deconstruction_state = HDD_PANEL_OPEN
+						to_chat(user, span_notice("You remove the last screw from [src]'s front panel. It drops, held only by wires."))
+						add_overlay("RD-server-hdd-panel-open")
+						return
+
+					to_chat(user, span_notice("You remove a screw from [src]'s front panel. Only [front_panel_screws] left..."))
+				return
+			if(istype(attacking_item, /obj/item/computer_hardware/hard_drive/cluster/hdd_theft))
+				to_chat(user, span_notice("You can't see a place to insert [src]."))
```

Comment:
```suggestion
				balloon_alert(user, "you can't find a place to insert it!")
```

Reply:
The others can be balloon alerts as well, though they already have their own visual feedback, so it's not necessary, plus I can see a reasonable case for looking back at them, I think.

Reply:
Changed.